File: src/main/java/com/sliusar/todolist/models/Todo.java
package com.sliusar.todolist.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.Date;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
@EqualsAndHashCode
@ToString(exclude = "schedule")
public class Todo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String title;
    private String body;
    private Date finalDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnore
    private Schedule schedule;


}

File: src/main/frontend/src/app/models/todo.ts
export class Todo {
  id: number | undefined;
  title: string;
  body: string;
  finalDate: Date;


  constructor(title: string, body: string, finalDate: Date) {
    this.title = title;
    this.body = body;
    this.finalDate = finalDate;
  }
}
